components/service/glean/src/main/java/mozilla/components/service/glean/EventMetricType.kt

```diff
@@ -109,4 +111,45 @@ data class EventMetricType(
             )
         }
     }
+
+    /**
+     * Tests whether a value is stored for the metric for testing purposes only
+     *
+     * @return true if metric value exists, otherwise false
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    fun testHasValue(
+        pingName: String =
+            if (sendInPings[0] == "default") { "metrics" } else { sendInPings[0] }
+    ): Boolean {
+        val snapshot = EventsStorageEngine.getSnapshot(pingName, false) ?: return false
+
+        return snapshot.any {
+            if (it.category.isEmpty()) {
```

Comment:
This is an engine implementation detail, should not leak outside here, as implementation might change. Maybe we should expose a test-only function for that in the engine?

Reply:
Maybe to make this less awkward, we could use a solution similar to the one for `CommonMetricData`: having an `internal val identifier: String` in `RecordedEventData` and using it here and in the engine.